code/addons/docs/src/preset.ts

```diff
@@ -58,7 +58,7 @@ async function webpack(
     mdxCompileOptions: {
       providerImportSource: '@storybook/addon-docs/mdx-react-shim',
       ...mdxPluginOptions.mdxCompileOptions,
-      remarkPlugins: [remarkSlug, remarkExternalLinks].concat(
+      remarkPlugins: [rehypeSlug, rehypeExternalLinks].concat(
         mdxPluginOptions?.mdxCompileOptions?.remarkPlugins ?? []
       ),
```

Comment:
as this is now rehype plugins and not remark plugins, they need to be added as such instead.

```suggestion
      rehypePlugins: [rehypeSlug, rehypeExternalLinks].concat(
        mdxPluginOptions?.mdxCompileOptions?.rehypePlugins ?? []
      ),
```